rpc/flipt/evaluation/evaluation.proto

```diff
@@ -67,6 +68,11 @@ message VariantEvaluationResponse {
   google.protobuf.Timestamp timestamp = 8;
 }
 
+message ErrorEvaluationResponse {
+  string flag_key = 1;
+  EvaluationReason reason = 2;
+}
```

Comment:
yeah that seems to track with https://openfeature.dev/specification/types#error-code , makes sense to me.

> We can basically namespace these things in their types so that they can only be populated with values that make sense

trying to remember if this is possible in proto3:

```proto
message ErrorEvaluationResponse {
  string flag_key = 1;
  enum ErrorReason {
    NONE_ERROR_REASON = 1;
    NOT_FOUND_ERROR_REASON = 2;
  } reason = 2;
}
```

So like you say, that enum only exists for that `ErrorEvaluationResponse`, otherwise maybe we should name it `EvaluationErrorReason` ?

Reply:
Yeah I think you can nest definitions inside one aother like that.

Reply:
talked about it on zoom, maybe we just pull the enum out like @GeorgeMac suggested, but call it `EvaluationErrorReason` ?